scripts/weekly-summary

```diff
@@ -0,0 +1,374 @@
+
+
+def run_curl(token, url, filename):
+    """
+    run_curl downloads an artifact file
+    """
+    with open(filename, "wb") as f:
+        c = pycurl.Curl()
+        headers = [
+           "Accept: application/vnd.github.v3+json",
+           f"Authorization: token {token}",
+        ]
+
+        options = {
+            pycurl.FOLLOWLOCATION:  1,
+            pycurl.MAXREDIRS:       5,
+            pycurl.CONNECTTIMEOUT:  30,
+            pycurl.TIMEOUT:         300,
+            pycurl.NOSIGNAL:        1,
+            pycurl.URL:             url,
+            pycurl.HTTPHEADER:      headers,
+            pycurl.WRITEDATA:       f
+        }
+        for k, v in options.items():
+            c.setopt(k, v)
+
+        try:
+            c.perform()
+            ok = (True, None)
+        except Exception as e:
+            print(f"ERROR: {e}", file=sys.stderr)
+            sys.stderr.flush()
+            ok = (False, e)
+        c.close()
+
+    return ok
+
+
+def download_artifacts(token, artifacts):
+    """
+    download_artifacts downloads the artifacts as uniquely named files
+    """
+    zipfiles = []
+    for a in artifacts:
+        updated_at = datetime.fromisoformat(a["updated_at"][:-1])
+        datename = a["name"]+updated_at.strftime("-%Y-%m-%d")
+        filename = datename + ".zip"
+        zipfiles.append((a["name"], datename, filename))
+        if os.path.exists(filename):
+            print(f"{filename} skipped, already downloaded")
+            continue
+
+        print(f"Fetching {filename}")
+        run_curl(token, a["archive_download_url"], filename)
+
+    return zipfiles
+
+
+def extract_logs(f):
+    """
+    extract_logs unzips the archive into a temporary directory
+
+    This directory is deleted when the object goes out of scope
+    """
+    tdir = tempfile.TemporaryDirectory(prefix="kstest-log-", dir="/var/tmp/")
+    with zipfile.ZipFile(f) as zf:
+        zf.extractall(tdir.name)
+
+    # Return the object so that the temporary directory isn't deleted yet
+    return tdir
+
+
+def rebuild_logs(tdir):
+    """
+    rebuild_logs recreates kstest.log with timestamps
+
+    It does this by appending all the individual kstest.log files, which do contain
+    timestamps, into a new kstest.log
+    """
+    # Remove the old log with no timestamps
+    kstest_log = os.path.join(tdir.name, "kstest.log")
+    os.unlink(kstest_log)
+
+    # Find all the test's kstest.log files and append them to kstest.log
+    with open(kstest_log, "w") as ksf:
+        for log in glob(os.path.join(tdir.name, "*", "kstest.log")):
+            with open(log) as f:
+                data = f.read(1024**2)
+                while data:
+                    ksf.write(data)
+                    data = f.read(1024**2)
+
+
+def pass_fail(tests):
+    """
+    pass_fail returns stats on the tests
+
+    It returns a tuple of total tests run, number passed, number failed
+    """
+    fail = 0
+    not_fail = 0
```

Comment:
I would rather name this `passed` instead of `not_fail`.

Also this is not necessary. When you are returning total tests runs that it's easy to compute that.

Reply:
passed is really close to 'pass' so I'll use success, since that's what it is looking for anyway.